artemis-server/src/main/java/org/apache/activemq/artemis/core/server/metrics/ActiveMQMetricsPlugin.java

```diff
@@ -26,4 +26,10 @@ public interface ActiveMQMetricsPlugin extends Serializable {
    ActiveMQMetricsPlugin init(Map<String, String> options);
 
    MeterRegistry getRegistry();
+
+   /**
+    * Allow additional binders to be bound at the right time
+    */
+   default void bind() {
```

Comment:
It's not clear why this new method is needed. Can you elaborate?

Reply:
sure. we have a timing issue in that ActiveMQMetricsPlugin::init is called before MetricsManager have had the chance to add the common tags.

any metrics created / registered before then will not have the "broker" tag. 

not sure there is an easy way to do that without breaking existing plugins?

Reply:
Are you adding your _own_ metrics in your plugin implementation?

Reply:
In my opinion the broker should simply support the processor and uptime metrics just like it does the JVM and Netty metrics. That way _every_ user could benefit from them and the plugin interface wouldn't need to change. I created https://issues.apache.org/jira/browse/ARTEMIS-4292 for this work.

Reply:
Back this change out and I'll merge it (assuming it passes the PR tests). Then I'll send a PR for ARTEMIS-4292.

Reply:
thanks. done.